### ISSUE TITLE
Add /etc/default/useradd management?

### ISSUE THREAD

**Issue by bschonec** [joshbeard/puppet-login_defs#17]
Since /etc/login.defs is part of the shadow-utils package, would it make sense to add /etc/default/useradd as well?

**Comment by joshbeard** [comment]
Hey @bschonec, I'm sorry for not following up on this sooner. I'm not opposed to optionally managing that here in a similar fashion. Both files are simple to manage as static files or templates, but I originally made this module so I could easily override things using Hiera.

On the other hand, the module is "login_defs", so I wouldn't want to be confusing. I think if we add it, it should be disabled by default. LMK what you think.

**Comment by bschonec** [comment]
My reasoning is that the shadow-utils RPM package contains both /etc/login.defs and /etc/default/useradd and it makes sense *to me* that your Puppet module would manage (all things?) contained in the shadow-utils RPM package.  I don't think it would be worth it to create a new Puppet module solely to manage a single file (useradd).  I'm happy to add it as I need the experience writing spec tests.

*I* have a need to manage /etc/defaults/useradd but that doesn't mean other people do as well.

**Comment by joshbeard** [comment]
@bschonec Yeah, I agree with that. If you want to put something together for that in this module, go for it. I think it'd be useful. But would you make it optional and turned off by default just to avoid surprises? I'm more cautious about it in this module since it's only ever managed the login.defs file for years without a ton of updates - so it's just kinda functioned as-is. I should've named the module "shadow" or something :)

As for the spec tests, this module may not be the best example of the "right way" to do it. Most of that is generated using the `pdk` tool, and you'll see the tests in this module are defined in the `matrix.yaml` file.

Let me know if I can help or if you have any questions. Thank you for your interest!
